common/kvstore/src/main/java/org/apache/spark/util/kvstore/LevelDBIterator.java

```diff
@@ -189,17 +189,7 @@ public synchronized void close() throws IOException {
       it.close();
       closed = true;
     }
-  }
-
-  /**
-   * Because it's tricky to expose closeable iterators through many internal APIs, especially
-   * when Scala wrappers are used, this makes sure that, hopefully, the JNI resources held by
-   * the iterator will eventually be released.
-   */
-  @SuppressWarnings("deprecation")
-  @Override
-  protected void finalize() throws Throwable {
```

Comment:
Why remove this? It may help clean up leaked iterators, though it's not by itself the mechanism to rely on.

Reply:
@srowen , thanks for comments. i realized `finalize` may not work as we expected, as `LevelDBIterator` objects are referenced in `LevelDB` till it's closed. You are right, it can help with leak. i updated PR to track `DBIterator` instead of `LevelDBIterator` in `LevelDB`, then function `finalize`  can still works as before.